addons/delivery/delivery.py

```diff
@@ -228,6 +253,24 @@ def get_price(self, cr, uid, id, order, dt, context=None):
         total = self.pool['res.currency'].compute(cr, uid, order.currency_id.id, order.company_id.currency_id.id, total, context=ctx)
         return self.get_price_from_picking(cr, uid, id, total,weight, volume, quantity, context=context)
 
+    def get_cost_from_picking(self, cr, uid, id, total, weight, volume, quantity, context=None):
```

Comment:
Ah, OK, certainly this is not "from picking", but you have used the same nomenclature. As this method is new, I suggest you another name like _get_cost_from_data_, which is more adjusted to the reality.